### PR TITLE
Improvements: Post-Processing revamp

### PR DESCRIPTION
**Pull request by ShaunFell**
This PR will add more documentation for the post processing routines

### CODE REVIEW

**Comment by ShaunFell** [comment]
This PR fixes #13

File: PostProcessing/Source/Common/__pycache__/Engine.cpython-37.pyc
Comment (Sbozzolo) [inline PostProcessing/Source/Common/__pycache__/Engine.cpython-37.pyc:1]: You should probably remove the `.pyc` files and add `.pyc` to your `.gitignore

File: PostProcessing/Source/Common/__init__.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+from .Engine import *
+from .Utils import *
+from .PlotSetup import *
+from .Plotter import *
```
Comment (Sbozzolo) [inline PostProcessing/Source/Common/__init__.py]: Some of your files are missing a final new line character

File: Tests/PostProcessing_test/main.py [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+            if not twod_success_status:
+                print("\n\n###############")
+                print("2D test failed!")
+                print("###############\n\n")
+            if not threed_success_status:
+                print("\n\n###############")
+                print("3D test failed!")
+                print("###############\n\n")
+
+        sys.exit(0 if success_status else 1)
```
Comment (Sbozzolo) [inline Tests/PostProcessing_test/main.py:251]:
```suggestion
        sys.exit(success_status)
```
should probably work

File: PostProcessing/Source/Common/__init__.py (hunk 1 above)
Comment (ShaunFell) [reply PostProcessing/Source/Common/__init__.py]: Ah, thanks for catching that. added a new commit fixing that

File: PostProcessing/Source/Common/__pycache__/Engine.cpython-37.pyc
Comment (ShaunFell) [reply PostProcessing/Source/Common/__pycache__/Engine.cpython-37.pyc:1]: Thats strange, because the .gitignore already contains those. I manually removed them from the remote, so that should fix it

File: PostProcessing/IntegralsPlot.py [hunk 3]
```diff
@@ -1,140 +1,64 @@
...
+    """
+    Main function to plot the data
+
+    Raises:
+        IOError: Data path not found
+        IOError: Data file not found
+        RuntimeError: No plot variables specified
+        SystemError: No plot variables specified
+        SystemError: Number of x limits does not match number of plot variables
+        SystemError: Number of linestyles does not match number of plot variables
```
Comment (Sbozzolo) [inline PostProcessing/IntegralsPlot.py]:
```suggestion
        RuntimeError: No plot variables specified
        RuntimeError: Number of x limits does not match number of plot variables
        RuntimeError: Number of linestyles does not match number of plot variables
```

File: PostProcessing/IntegralsPlot.py [hunk 4]
```diff
@@ -1,140 +1,64 @@
...
     # get the file object
     data_obj = get_data_dataframe(data_filename_abs)
 
     #get the plot variables
-    plot_variables = config["VariableData"].get("plot_variables", "").split()    
-    #check the plot variables are non-empty
-    if not plot_variables:
-        raise SystemError("No plot variables specified.")
+    plot_variables = config["VariableData"].get("plot_variables", "").split()
+    if len(plot_variables) == 0:
```
Comment (Sbozzolo) [inline PostProcessing/IntegralsPlot.py]: If you want, you can just write
```suggestion
    if plot_variables:
```
An empty list evaluates to false.

File: PostProcessing/IntegralsPlot.py [hunk 5]
```diff
@@ -143,26 +67,25 @@ def main():
...
-
-
-
 if __name__ == "__main__":
-    main()
\ No newline at end of file
+    main()
+elif __visit_script_file__ == __visit_source_file__:
+    raise RuntimeError("This file should not be run with VisIt, only Python!")
+
```
Comment (Sbozzolo) [inline PostProcessing/IntegralsPlot.py]: This file doesn't end with a newline

File: PostProcessing/IntegralsPlot.py [hunk 6]
```diff
@@ -143,26 +67,25 @@ def main():
         linestyles = None
 
     if not len(linestyles) == len(plot_variables) and linestyles != None:
```
Comment (Sbozzolo) [inline PostProcessing/IntegralsPlot.py]: It's a little hard to follow these conditionals

also, comparisons with `None` should be done with `is` instead of `==`

File: PostProcessing/IntegralsPlot.py (hunk 3 above)
Comment (ShaunFell) [reply PostProcessing/IntegralsPlot.py]: Thanks, forgot to change the docstrings for that

File: PostProcessing/IntegralsPlot.py (hunk 4 above)
Comment (ShaunFell) [reply PostProcessing/IntegralsPlot.py]: Ah, yea thats right. I've been using so much C++ i forgot about those nice pythonic things. Changed in newest commit to this PR

File: PostProcessing/IntegralsPlot.py (hunk 6 above)
Comment (ShaunFell) [reply PostProcessing/IntegralsPlot.py]: fixed in newest commit https://github.com/ShaunFell/GRBoondi/pull/18/commits/edcbb92ca1e7867895adebd7bbcd814aab825162

File: PostProcessing/3DPlotParams.ini [hunk 7]
```diff
@@ -5,6 +5,7 @@
 hdf5_path = /home/hd/hd_hd/hd_pb293/WS_GRChombo/testing/GRMilijun_ProcaKerrBH_23320250/hdf5
```
Comment (Sbozzolo) [inline PostProcessing/3DPlotParams.ini:5]: This looks like an hardcoded path that users won't have access to

File: PostProcessing/Source/Common/Utils.py [hunk 8]
```diff
@@ -0,0 +1,378 @@
...
+    filename_prefix = os.path.join(config["Header"]["hdf5_path"], config["Header"]["plot_header"])
+
+    #get list of file using regex
+    files = glob.glob(filename_prefix+ "*.3d.hdf5")
+
+    #filter the file list to ensure regex-captured files are actual plot files
+    plot_files = [x for x in files if config["Header"]["plot_header"] in x]
+
+    #ensure plot files exist in provided directory
+    if len(plot_files) == 0:
```
Comment (Sbozzolo) [inline PostProcessing/Source/Common/Utils.py]: Same here, if you want to remove the `len == 0`, you can do it

File: PostProcessing/Source/Common/Utils.py [hunk 9]
```diff
@@ -0,0 +1,378 @@
...
+    """
+
+    #path to hdf5 files plus plot file header string
+    filename_prefix = os.path.join(config["Header"]["hdf5_path"], config["Header"]["plot_header"])
+
+    #get list of file using regex
+    files = glob.glob(filename_prefix+ "*.3d.hdf5")
+
+    #filter the file list to ensure regex-captured files are actual plot files
+    plot_files = [x for x in files if config["Header"]["plot_header"] in x]
```
Comment (Sbozzolo) [inline PostProcessing/Source/Common/Utils.py]: You could also use the `filter` function to express your intent more clearly

File: PostProcessing/VisitPlot.py [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+		make_movie(config, plotvar)
+
+		print("I've finished!")
+	
+	#cleanup	
+	os.remove("./visitlog.py")
+	Close_Database(config) #includes error handling
+
+	print("All Done!")
+	sys.exit(0)
```
Comment (Sbozzolo) [inline PostProcessing/VisitPlot.py:77]: Is this needed?

**Review by Sbozzolo — APPROVED** [review]
Thank you!

File: PostProcessing/3DPlotParams.ini (hunk 7 above)
Comment (ShaunFell) [reply PostProcessing/3DPlotParams.ini:5]: the *.ini files are parameter files created by the user, as explained [here](https://github.com/ShaunFell/GRBoondi/wiki/Visualization#postprocessing)

File: PostProcessing/Source/Common/Utils.py (hunk 9 above)
Comment (ShaunFell) [reply PostProcessing/Source/Common/Utils.py]: Ah, thats probably better. added in latest commit
